client/lib/signup/flow-controller.ts

```diff
@@ -123,6 +124,22 @@ export default class SignupFlowController {
 		this._resetStoresIfUserHasLoggedIn(); // reset the stores if user has newly authenticated
 		this._resetSiteSlugIfUserEnteredAnotherFlow(); // reset the site slug if user entered another flow
 
+		// If we have access to the siteId, then make sure we've also loaded the siteSlug into
+		// the dependency store, because some code depends on the slug instead of the id.
+		if (
+			this._flow.providesDependenciesInQuery?.includes( 'siteId' ) &&
+			options.providedDependencies[ 'siteId' ] &&
+			! options.providedDependencies[ 'siteSlug' ]
+		) {
+			const siteSlug = getSiteSlug(
+				this._reduxStore.getState(),
+				options.providedDependencies[ 'siteId' ]
```

Comment:
If we already have `siteId`, can we fetch `siteSlug` by `siteId`? Or is there any problem? I feel like using `siteId` as `siteSlug` might be confused easily.

Reply:
Yes that's what the `getSiteSlug()` is doing here, we're fetching `siteSlug` by `siteId`. I might not quite understand what you mean though.

Reply:
Oops...I saw `siteId` there and misunderstood this one 😅